zboss/CMakeLists.txt

```diff
@@ -38,14 +43,42 @@ if ((NOT DEFINED ZEPHYR_PLATFORM_NCS_MODULE_DIR) OR
                     "(${ZBOSS_LIB_PATH} doesn't exist.)")
   endif()
 
+  target_include_directories(zboss INTERFACE
+    include
+    include/zcl
+    include/ha
+    include/se
+    include/osif
+  )
+
+  # Define the ZBOSS imported library, that contains only static libraries
   if (CONFIG_ZIGBEE_ROLE_COORDINATOR OR CONFIG_ZIGBEE_ROLE_ROUTER)
-    zephyr_link_libraries(${ZBOSS_LIB_PATH}/libzboss.a)
-    toolchain_cc_imacros(${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config.h)
+    zephyr_library_import(zboss-libraries ${ZBOSS_LIB_PATH}/libzboss.a)
+    set(LIBZBOSS_CONFIG_FILE "libzboss_config.h")
   elseif (CONFIG_ZIGBEE_ROLE_END_DEVICE)
-    zephyr_link_libraries(${ZBOSS_LIB_PATH}/libzboss.ed.a)
-    toolchain_cc_imacros(${CMAKE_CURRENT_SOURCE_DIR}/include/osif/libzboss_config.ed.h)
+    zephyr_library_import(zboss-libraries ${ZBOSS_LIB_PATH}/libzboss.ed.a)
+    set(LIBZBOSS_CONFIG_FILE "libzboss_config.ed.h")
   else()
     message( FATAL_ERROR "Unsupported Zigbee role. Exiting." )
   endif()
 
+  # Link with Zephyr's default target
+  zephyr_link_libraries(zboss)
```

Comment:
Same as above.

Reply:
Fixed. Now application links to zboss, which links to zboss-extensions, as well as zboss-extensions C files, which have autogenerated name.